src/main/kotlin/uk/gov/dluhc/printapi/rds/entity/Address.kt

```diff
@@ -0,0 +1,83 @@
+package uk.gov.dluhc.printapi.rds.entity
+
+import org.hibernate.Hibernate
+import org.hibernate.annotations.GenericGenerator
+import org.hibernate.annotations.Type
+import org.hibernate.annotations.UpdateTimestamp
+import org.springframework.data.annotation.LastModifiedBy
+import org.springframework.data.jpa.domain.support.AuditingEntityListener
+import uk.gov.dluhc.printapi.rds.repository.UUIDCharType
+import uk.gov.dluhc.printapi.rds.repository.UseExistingOrGenerateUUID
+import java.time.Instant
+import java.util.UUID
+import javax.persistence.Entity
+import javax.persistence.EntityListeners
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.Table
+import javax.persistence.Version
+import javax.validation.constraints.NotNull
+import javax.validation.constraints.Size
+
+@Table
+@Entity
+@EntityListeners(AuditingEntityListener::class)
+class Address(
+    @Id
+    @Type(type = UUIDCharType)
+    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "UUID")
+    @GenericGenerator(name = "UUID", strategy = UseExistingOrGenerateUUID.NAME)
+    var id: UUID? = null,
+
+    @NotNull
+    @Size(max = 255)
+    var street: String? = null,
+
+    @NotNull
+    @Size(max = 10)
+    var postcode: String? = null,
+
+    @Size(max = 255)
+    var `property`: String? = null,
+
+    @Size(max = 255)
+    var locality: String? = null,
+
+    @Size(max = 255)
+    var town: String? = null,
+
+    @Size(max = 255)
+    var area: String? = null,
+
+    @Size(max = 12)
+    var uprn: String? = null,
+
+    @NotNull
+    @UpdateTimestamp
+    var dateCreated: Instant? = null,
```

Comment:
I am aware we are doing something similar in vca with `dateCreated as @UpdateTimestamp ` 
Should we use @CreationTimestamp instead (not a blocker, just a question)

Reply:
I would vote for 2 separate fields - dateCreated and lastUpdated with the relevant annotations 👍

Reply:
Having said that, it's different for immutable entities. The `Address` entity in RCA just has a `dateCreated` field with the `@CreationTimestamp` annotation:

```
    @NotNull
    @CreationTimestamp
    var dateCreated: Instant? = null,
```

Mutable objects also have `updatedAt` and `version`:
```
    @NotNull
    @UpdateTimestamp
    var updatedAt: Instant? = null,

    @Version
    var version: Long? = null,
```

Reply:
Yes, C&P from VAC.. will use created annotation 👍